### PR TITLE
Default Configuration Paths

### PR DESCRIPTION
**Pull request by samlikins**
Addresses issue #477

### CODE REVIEW

File: cement/core/foundation.py [hunk 1]
```diff
@@ -1265,13 +1265,17 @@ def _setup_config_handler(self):
         if self._meta.config_files is None:
             self._meta.config_files = [
                 os.path.join('/', 'etc', label, '%s%s' % (label, ext)),
+                os.path.join(fs.HOME_DIR, '.config', label, '%s%s' % (label, ext)),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/<app_label>.conf

File: cement/core/foundation.py [hunk 2]
```diff
@@ -1265,13 +1265,17 @@ def _setup_config_handler(self):
...
+                os.path.join(fs.HOME_DIR, '.config', label, 'config'),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/config

File: cement/core/foundation.py [hunk 3]
```diff
@@ -1265,13 +1265,17 @@ def _setup_config_handler(self):
...
+                os.path.join(fs.HOME_DIR, '.config', '%s%s' % (label, ext)),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>.conf

File: cement/core/foundation.py [hunk 4]
```diff
@@ -1265,13 +1265,17 @@ def _setup_config_handler(self):
...
                 os.path.join(fs.HOME_DIR, '.%s%s' % (label, ext)),
                 os.path.join(fs.HOME_DIR, '.%s' % label, 'config'),
             ]
 
         if self._meta.config_dirs is None:
             self._meta.config_dirs = [
                 os.path.join('/', 'etc', label, 'conf.d'),
+                os.path.join(fs.HOME_DIR, '.config', label, 'conf.d'),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/conf.d

File: cement/core/foundation.py [hunk 5]
```diff
@@ -1345,6 +1349,7 @@ def _setup_plugin_handler(self):
         if self._meta.plugin_config_dirs is None:
             self._meta.plugin_config_dirs = [
                 '/etc/%s/plugins.d/' % self._meta.label,
+                os.path.join(fs.HOME_DIR, '.config', label, 'plugins.d'),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/plugins.d

File: cement/core/foundation.py [hunk 6]
```diff
@@ -1356,6 +1361,7 @@ def _setup_plugin_handler(self):
         # plugin dirs
         if self._meta.plugin_dirs is None:
             self._meta.plugin_dirs = [
+                os.path.join(fs.HOME_DIR, '.config', label, 'plugins'),
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/plugins

File: cement/core/foundation.py [hunk 7]
```diff
@@ -1392,6 +1398,7 @@ def _setup_output_handler(self):
         if self._meta.template_dirs is None:
             self._meta.template_dirs = []
             paths = [
+                os.path.join(fs.HOME_DIR, '.config', label, 'templates')
```
Comment (samlikins) [inline cement/core/foundation.py]: ~/.config/<app_label>/templates

**Comment by derks** [comment]
This was merge in manually to `portland` (Cement 3).

**Pull request closed without merging**